azure-pipelines.yml

```diff
@@ -10,168 +10,120 @@ variables:
 pool:
   vmImage: 'Ubuntu 16.04'
 
-jobs:
-  - job: Build
+stages:
+  - stage: Build
     pool: 'Self Host Ubuntu'
-    steps:
-      - template: azure-pipelines.tools.yml
-
-      - script: |
-          yarn
-        displayName: yarn
-
-      - script: |
-          yarn checkchange
-        displayName: check change
-
-      - script: |
-          yarn build
-        displayName: build
-
-      - task: ArchiveFiles@2
-        inputs:
-          rootFolderOrFile: $(Build.Repository.LocalPath)
-          # Include root folder so archive can just be extracted into $(Pipeline.Workspace)
-          includeRootFolder: true
-          archiveType: 'tar'
-          tarCompression: 'gz'
-          archiveFile: $(Build.ArtifactStagingDirectory)/build.tar.gz
-          replaceExistingArchive: true
-
-      - publish: $(Build.ArtifactStagingDirectory)
-        artifact: Build
-
-      - task: DeleteFiles@1
-        inputs:
-          SourceFolder: $(Build.SourcesDirectory)
-          Contents: '**/*'
-
-  - job: Validate
+    jobs:
+      - job: BuildJob
+        steps:
+          - template: azure-pipelines.tools.yml
+
+          - script: |
+              yarn
+            displayName: yarn
+
+          - script: |
+              yarn checkchange
+            displayName: check change
+
+          - script: |
+              yarn build
+            displayName: build
+
+          - task: ArchiveFiles@2
+            inputs:
+              rootFolderOrFile: $(Build.Repository.LocalPath)
+              # Include root folder so archive can just be extracted into $(Pipeline.Workspace)
+              includeRootFolder: true
+              archiveType: 'tar'
+              tarCompression: 'gz'
+              archiveFile: $(Build.ArtifactStagingDirectory)/build.tar.gz
+              replaceExistingArchive: true
+
+          - publish: $(Build.ArtifactStagingDirectory)
+            artifact: Build-PR-$(Build.BuildNumber)
+
+          - script: |
+              az --version
+              az extension add -n azure-devops
+              echo $(buildQueuePat) | az devops login --organization https://uifabric.visualstudio.com
+              az devops configure --defaults organization=https://uifabric.visualstudio.com project="fabricpublic" --use-git-aliases true
+            env:
+              AZURE_DEVOPS_EXT_PAT: $(buildQueuePat)
```

Comment:
Does this have any security implications?

Reply:
We have a token here that triggers builds. So someone could potentially kick off more build jobs within the fabricpublic project which they already can by forking & submitting PRs.